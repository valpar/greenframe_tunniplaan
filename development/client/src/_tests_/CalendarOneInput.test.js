import React from 'react';
import { render, fireEvent, waitFor, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'; //snapshot renderer
import CalendarOneInput from '../components/UI/Calendar/CalendarOneInput';

// Mock the onChange prop function
const mockOnChange = jest.fn();

afterEach(() => {
    cleanup();
});

describe('CalendarOneInput', () => {

    //snapshoti test on, teha ka unit testid
    it("matches snapshot", () => {
        // loob snapshoti
        const tree = renderer.create(<CalendarOneInput/>).toJSON();
        // console.log(tree);
        // v√µrdleb testis loodud snapshotiga, kui komponenti muudetakse, siis test feilib, "u" shortcut keyga saab uue snapshoti luua
        expect(tree).toMatchSnapshot();
    });
        
      
      
});
