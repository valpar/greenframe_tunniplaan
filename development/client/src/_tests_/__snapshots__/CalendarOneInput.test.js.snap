// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarOneInput matches snapshot 1`] = `
<div
  className="relative w-full"
>
  <div
    className="relative flex flex-col w-full"
  >
    <input
      autoComplete="off"
      className="p-2 w-full border border-borderGray  rounded-none font-normal focus:outline-none focus:border-black undefined"
      name="date"
      onChange={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      readOnly={false}
      type="text"
      value=""
    />
  </div>
</div>
`;
